### PR TITLE
Filters enhancement

### PR DESCRIPTION
**Pull request by marktriggs**

### CODE REVIEW

**Comment by marktriggs** [comment]
No mergy yet!  Still making some changes

File: frontend/app/helpers/enumeration_helper.rb [hunk 1]
```diff
@@ -1,9 +1,21 @@
 module EnumerationHelper
 
-  def enumeration_simple_filter_params( relationships, value ) 
-    pattern = %r{(\+|\-|\&\&|\|\||\!|\(|\)|\{|\}|\[|\]|\^|\"|\~|\*|\?|\ |\:|\\)}
-    value.gsub!(pattern) { |match| '\\' + match } 
-    relationships.map { |rel| "#{rel}_enum_s:#{value}"  }.join(" OR ") 
+  def enumeration_advanced_query(relationships, value)
```
Comment (marktriggs) [inline frontend/app/helpers/enumeration_helper.rb]: use new builder here

File: frontend/app/models/search.rb [hunk 2]
```diff
@@ -20,11 +23,35 @@ def self.all(repo_id, criteria)
...
+        'op' => 'AND',
+        'type' => 'text',
+        'field' => filter.keys[0],
+        'value' => filter.values[0],
+      }
+    end
+
+    return if queries.empty?
+
+    result = AdvancedQueryBuilder.build_query_from_form(queries)
```
Comment (marktriggs) [inline frontend/app/models/search.rb]: can we use the new builder for this too?

**Comment by marktriggs** [comment]
OK, ready for review :pig:

**Comment by payten** [comment]
🍄 🌝

**Pull request merged**